nio/client/async_client.py

```diff
@@ -337,14 +337,11 @@ def add_response_callback(
                 response_data
             )
 
-            if isinstance(response, ErrorResponse):
+            if isinstance(response, ErrorResponse) and response.retry_after_ms:
                 await self.run_response_callbacks([response])
```

Comment:
I overlooked this the first time round. Response callbacks exist so that users can react to requests/responses that the client makes on behalf of them. This way they can react to responses that they don't make on their own.

A user doesn't need to react to an error if the client will handle it transparently for them, if they would react to this error and retry the response on their own we would have duplicate requests going out.

Do you have a use-case where this is needed?

Reply:
A common way of getting the error is spamming in a room, so this makes me able to show a warning in my client and/or force a cooldown on the user before sending more messages.